evm/vm/state_transition_helper.py

```diff
@@ -0,0 +1,47 @@
+import copy
+import logging
+import traceback
+
+
+logger = logging.getLogger("evm.research.state_transition_helper")
+
+
+def apply_transaction(transaction, block, prev_vm, chaindb):
+    """
+    Apply transaction to the block, return the result with a new block object
+
+    Original Goal: apply_transaction(stateobj, db, blockdata, transaction)
+    -> state_obj', reads, writes
+    """
+    # TODO: try to use simpler StateObj object instead of VM as the
+    # state transition container?
+
+    # init vm
+    vm = copy.deepcopy(prev_vm)
+    vm.chaindb = chaindb
+
+    # result
+    success = True
+    reads = None
+    writes = None
+
+    try:
+        vm.set_stateless(True)
+        computation = vm.apply_transaction_to_block(
+            transaction,
+            block,
+            vm.chaindb,
+        )
+        vm.set_stateless(False)
+        if computation.is_error:
+            success = False
+    except Exception as e:
+        # FIXME: any specific exception type in this function?
+        logger.error(
+            "Unexpected error when handling remote msg: {}".format(traceback.format_exc()))
+        success = False
+
+    reads = vm.reads
+    writes = vm.writes
+
+    return success, reads, writes, vm.block
```

Comment:
Instead of success/failure can we return the computation object itself?  Reasoning is that it should still be a fine representation of success/failure but it also allows for deeper inspection of what happened during computation which I think is valuable and was one of the things missing in pyethereum that I really wanted.  Things like accessing the `output` of computation or really anything else that's exposed through the computation object.

Reply:
Yes, I will change to returning `computation`.